pkg/logql/engine.go

```diff
@@ -278,10 +278,20 @@ func (q *query) Eval(ctx context.Context) (promql_parser.Value, error) {
 
 	switch e := expr.(type) {
 	case syntax.SampleExpr:
+		// Selector error will be handled later.
+		selector, _ := e.Selector()
+		if err := q.validateMatchers(ctx, tenants, selector.Matchers()); err != nil {
```

Comment:
I'm wondering if we should use `MatcheGroups` instead. If I understand #8670 correctly we can have queries like this

```
count_over_time({job="foo"}[5m]) / count_over_time({pod="bar"}[5m] offset 10m)
```

What would `Matchers` return?